File: react-native-meridian-maps/android/src/main/java/com/meridianmaps/TriggersReceiver.java
package com.meridianmaps;

import android.content.Context;
import android.content.Intent;

import com.arubanetworks.meridian.editor.EditorKey;
import com.arubanetworks.meridian.triggers.TriggersBroadcastReceiver;
import com.squareup.otto.Bus;
import com.squareup.otto.ThreadEnforcer;

import androidx.annotation.NonNull;
import android.util.Log;

public class TriggersReceiver extends TriggersBroadcastReceiver {

  @Override
  protected void onReceive(Context context, Intent intent, String title) {
    String appId = intent.getStringExtra("APP_KEY");
    if (appId == null) {
      Log.e("TriggerReceiver onReceive", "No app key in trigger intent");
      return;
    }
    EditorKey appKey = EditorKey.forApp(appId);
    TriggerEvent.getInstance().post(new TriggerEvent(intent, title));
  }

  protected static class TriggerEvent extends Bus {
    private static final Bus BUS = new Bus(ThreadEnforcer.MAIN);

    private final String name;
    private final int rssi;
    private final int major;
    private final int minor;
    private final float x;
    private final float y;
    private final String mapId;
    private final String mac;

    TriggerEvent(@NonNull Intent intent, @NonNull String triggerName) {
      this.name = triggerName;
      this.rssi = TriggersBroadcastReceiver.getRSSI(intent);
      this.major = TriggersBroadcastReceiver.getMajor(intent);
      this.minor = TriggersBroadcastReceiver.getMinor(intent);
      this.x = TriggersBroadcastReceiver.getX(intent);
      this.y = TriggersBroadcastReceiver.getY(intent);
      this.mapId = TriggersBroadcastReceiver.getMapId(intent);
      this.mac = TriggersBroadcastReceiver.getMacAddress(intent);
    }

    public String getName() {
      return name;
    }

    public int getRSSI() {
      return rssi;
    }

    public int getMajor() {
      return major;
    }

    public int getMinor() {
      return minor;
    }

    public float getX() {
      return x;
    }

    public float getY() {
      return y;
    }

    public String getMapId() {
      return mapId;
    }

    public String getMacAddress() {
      return mac;
    }

    public static Bus getInstance() {
      return BUS;
    }
  }
}

File: react-native-meridian-maps/android/src/main/java/com/meridianmaps/TriggersReceiver.kt
package com.meridianmaps

import android.content.Context
import android.content.Intent
import com.arubanetworks.meridian.triggers.TriggersBroadcastReceiver
import com.squareup.otto.Bus
import com.squareup.otto.ThreadEnforcer
import kotlin.jvm.JvmStatic

class TriggersReceiver : TriggersBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent, triggerName: String) {
        val appKey = getAppKey(intent)
        if (appKey != null && appKey.id.equals(Application.APP_KEY.id, ignoreCase = true)) {
            TriggerEvent.getInstance().post(TriggerEvent(intent, triggerName))
        }
    }

    /**
     * Create an OTTO event to update the trigger fragment
     */
    class TriggerEvent internal constructor(intent: Intent, triggerName: String) : Bus() {
        private val BUS: Bus = Bus(ThreadEnforcer.MAIN)
        val name : String
        val rssi : Int
        val major : Int
        val minor : Int
        val x : Float
        val y : Float
        val mapId : String
        val mac : String

        companion object {
            private val BUS: Bus = Bus(ThreadEnforcer.MAIN)
            @JvmStatic
            public fun getInstance(): Bus {
                return TriggerEvent.BUS
            }
        }

        init {
            this.name = triggerName
            this.rssi = getRSSI(intent)
            this.major = getMajor(intent)
            this.minor = getMinor(intent)
            this.x = getX(intent)
            this.y = getY(intent)
            this.mapId = getMapId(intent)
            this.mac = getMacAddress(intent)
        }
    }
}
